crates/shared/src/price_estimation/oneinch.rs

```diff
@@ -19,6 +19,7 @@ impl OneInchPriceEstimator {
             return Err(PriceEstimationError::UnsupportedOrderType);
         }
 
+        // If applicable, keep these parameters synchronized with OneInchSolver swap parameters!
```

Comment:
Instead of manually synchronizing can we maybe add something like:
```
impl SellOrderQuoteQuery {
    pub fn with_default_options(sell_token, buy_token, in_amount, protocols) -> Self {
        // ...
    }
}
```

To construct a `SellOrderQuoteQuery` with the default options? This way it there is no manual requirement to keep the two files in sync.

Reply:
To have a "single source of truth" we would have to have a struct storing the defaults for settings which are supported in both `SwapQuery` and `SellOrderQuoteQuery` and construct the respective types from it.
If it's good enough to manually keep code synchronized within a single file I would add `with_default_options` function to both structs.

Reply:
Ah! They are different structs.

One other possibility would be to:
```rust
struct SellOrderQuoteQuery {
  things: Things,
  common: CommonQueryParameters,
}

struct SwapQuery {
  other_things: OtherThings,
  common: CommonQueryParameters,
}

struct CommonQueryParameters {
  common_things: CommonThings,
}

impl Default for CommonQueryParameters {
  fn default() -> Self {
    // ...
  }
}
```

This way we can keep track of the common parameter defaults that we used. I'm always wary of duplicating "magic values" as, in my experience, they very quickly get out of sync despite comments urging them to stay in sync.